google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionImpl.java

```diff
@@ -252,29 +246,136 @@ public BigQueryResult executeSelect(
       throw new BigQuerySQLException(e.getMessage(), e, e.getErrors());
     }
   }
+  /**
+   * Execute a SQL statement that returns a single ResultSet and returns a ListenableFuture to
+   * process the response asynchronously.
+   *
+   * <p>Example of running a query.
+   *
+   * <pre>
+   * {
+   *   &#64;code
+   *   ConnectionSettings connectionSettings =
+   *        ConnectionSettings.newBuilder()
+   *            .setRequestTimeout(10L)
+   *            .setMaxResults(100L)
+   *            .setUseQueryCache(true)
+   *            .build();
+   *   Connection connection = bigquery.createConnection(connectionSettings);
+   *   String selectQuery = "SELECT corpus FROM `bigquery-public-data.samples.shakespeare` GROUP BY corpus;";
+   * ListenableFuture<ExecuteSelectResponse> executeSelectFuture = connection.executeSelectAsync(selectQuery);
+   * ExecuteSelectResponse executeSelectRes = executeSelectFuture.get();
+   *
+   *  if(!executeSelectRes.getIsSuccessful()){
+   * throw executeSelectRes.getBigQuerySQLException();
+   * }
+   *
+   *  BigQueryResult bigQueryResult = executeSelectRes.getBigQueryResult();
+   * ResultSet rs = bigQueryResult.getResultSet();
+   * while (rs.next()) {
+   * System.out.println(rs.getString(1));
+   * }
+   *
+   * </pre>
+   *
+   * @param sql a static SQL SELECT statement
+   * @return a ListenableFuture that is used to get the data produced by the query
+   * @exception BigQuerySQLException if a database access error occurs
+   */
+  @BetaApi
+  @Override
+  public ListenableFuture<ExecuteSelectResponse> executeSelectAsync(String sql)
+      throws BigQuerySQLException {
+    return getExecuteSelectFuture(sql, null);
+  }
+
+  /** This method calls the overloaded executeSelect(...) methods and returns a Future */
+  private ListenableFuture<ExecuteSelectResponse> getExecuteSelectFuture(
+      String sql, List<Parameter> parameters, Map<String, String>... labels) {
+    ExecutorService execService =
+        Executors.newFixedThreadPool(
+            2); // two fixed threads. One for the async operation and the other for processing the
+    // callback
+    ListeningExecutorService lExecService = MoreExecutors.listeningDecorator(execService);
+    ListenableFuture<ExecuteSelectResponse> executeSelectFuture =
+        lExecService.submit(
+            () -> {
+              if (parameters == null) {
```

Comment:
is there value in doing this check?  does executeSelect not properly deal with null params?

Reply:
Thanks for pointing this our @shollyman , `null` check isn't required here as the underlying method deals with it. Replaced it with a call to `executeSelect(  sql,  parameters,  labels)`. Also added exception handling to pass back any exception which might have occurred.